### ISSUE TITLE
Why reciprocal license restrictions in "Guide for Using Open Source Software"?

### ISSUE THREAD

**Issue by kfogel** [canada-ca/open-source-logiciel-libre#62]
In the [Guide for Using Open Source Software](https://github.com/canada-ca/open-source-logiciel-libre/blob/master/en/guides/using-open-source-software.md), in the section [Verify Open Source Software License](https://github.com/canada-ca/open-source-logiciel-libre/blob/master/en/guides/using-open-source-software.md#verify-open-source-software-licence), it says that applications that are going to be modified can't use a "strong reciprocal" license if they're web apps nor *any* reciprocal license if they're going to be distributed externally.

This raises several questions/issues:

* In general, one can never predict with certainty whether one will have to make modifications to an open source software application one is deploying.  One may start out intending to make no modifications, and discover one year down the road that modifications are necessary (furthermore, one cannot predict whether those modifications would be of the sort that should be submitted upstream, rather than merely published, and whether upstream would accept them if they were submitted).

* Since the [Guide for Publishing Open Source Code](https://github.com/canada-ca/open-source-logiciel-libre/blob/master/en/guides/publishing-open-source-code.md) says that work should happen in the open by default, in principle all modifications are intended for external distribution anyway (except for those which meet one of the rare exception conditions stated, but those do not concern us here).

What this all adds up to is that the [Guide for Using Open Source Software](https://github.com/canada-ca/open-source-logiciel-libre/blob/master/en/guides/using-open-source-software.md) effectively prohibits GC from using any reciprocally-licensed (i.e., copyleft / GPL'd) software at all.

Presumably this was not the intent?

What is the purpose of discouraging contact with reciprocal licenses, as specified in points 3 and 4 of the section "Verify Open Source Software Licence", anyway?  Or am I just misreading the text somehow?

FWIW, the *publication* guide (a separate document) does not similarly discourage use of reciprocal licenses -- it just lists them as an option along with other types of licenses.  Unless I've misunderstood something, the interaction between those two policies means that GC could easily end up publishing open source software that GC itself cannot re-use.

**Comment by smellems** [comment]
It's supposed to represent a decision tree.  So you only get to points 3 and 4 if you answer Yes to `2. Are there any reasons that would prevent the release of the modified source code?`.  In most cases you should be able to answer No.

Would the current situation or your department policies prevent publication of source code of modified OSS?  If yes, and it's a Web application, don't use strong reciprocal (AGPL).  If yes, and your going to be "distributing" the resulting software, use only permissive licences.

**Comment by gcharest** [comment]
Yes, I think we need to work some kind of graphics... The decision tree is
meant to go from 80% scenarios you should be able to use OSS under any open
source licence and then break out the remaining 20% scenarios.

Le mar. 7 mai 2019 07 h 55, Sébastien Lemay <notifications@github.com> a
écrit :

**Comment by gcharest** [comment]
We probably should add some nuance as well to help teams realize what would constitute a reason that the source code could not be published, linking to the conditions we mentioned in the white paper and publishing OSS.

There's really little reason why we should modify existing OSS and introduce code that shouldn't be released itself. 

It may sound overkill, but we may need to add some additional disambiguation around difference of source code and software deployed. I still have to explain the difference to people I should not... (That's not just government by the way - it's a bit scary at times)

**Comment by kfogel** [comment]
Ah, I see now how it represents a decision tree, but that wasn't clear to me when I first read it through, FWIW.  A graphic would be great.

**Comment by gcharest** [comment]
Absolutely! Yes, definitely need graphics... Also, clearer introduction might help in the meantime!

**Comment by ShadeWyrm** [comment]
![Reciprocal vs Permissive Licenses in GC](https://user-images.githubusercontent.com/43480681/66332730-5f075300-e903-11e9-9e7c-b09ad0fa2cef.PNG)

@gcharest  / @kfogel / @smellems - hi all! Can I get your feedback on the above, and if you feel it more clearly explains the thought process/flow?

**Comment by smellems** [comment]
This is nice, but its not about reciprocal vs permissive, I think.

Its about what OSS licences are acceptable for use by the GC.  Not releasing or even contributing back.  For example `but we recommend permissive`.  We can't recommend in this case because we're not picking the license of OSS projects being used only saying if it can be used.

I think it should start with
> When using OSS **without modifications**, all software licensed under an OSI approved license or a FSF free software licence is considered OSS and can be used.

Same for when you will make modifications and can share them.

If you can't share modifications the follow the questions..

For the yellow boxes I propose the following changes to be less about permissive vs reciprocal (from the top)

1. Source code should be shared if it does not include the things it should not include.
2. Consider contributing back to upstream communities.
3. Strong reciprocal licences consider access through a network (like the Internet) distribution and the modified source code **must** be made available.
4. When distributing modified versions of OSS using reciprocal licences, the modified source code **must** be made available.

**Comment by gcharest** [comment]
Ah yes, the decision tree was more about the software used inbound. A decision tree should probably be done for licences applied to GC projects :thinking: 

For example, see this graphic from Heather Meeker's book Open (Source) for Business:
/home/gcharest/Documents/github/sara-sabr/util-presentation/assets/apache2-gpl-error.png

**Comment by ShadeWyrm** [comment]
![UOSS](https://user-images.githubusercontent.com/43480681/68024940-e7b6ac00-fc81-11e9-8c4c-2cc55758267e.PNG)

How is this in comparison?
